docs/apis/subsystems/communication/index.md

```diff
@@ -0,0 +1,244 @@
+not be available for any checks (capability, role etc.) after deleted.
+
+:::info
+
+This is an experimental feature. To use this feature, you must enable the experimental feature flag for Communication in the site administration.
+
+:::
+
+### Loading a communication instance
+
+To load a communication instance, you must use the communication\api::load_by_instance() method. This method will return a communication instance which can be
+used to interact with the communication API and its related methods. The below example shows how to load a communication instance for a course, where the
+context is the actual course context object, component is the component name (core_course), instance type is the custom string which can change according to
+the usage and instance id is the course id. The constructor of the public api is private and hence the only way to load an instance is through
+the load_by_instance() method. The provider is the name of the provider plugin which is used to load the instance. It's optional, if not provided, it will load
+the enabled provider for that instance.
+
+```php
+$communication = \core_communication\api::load_by_instance(
+            context: $context,
+            component: 'core_course',
+            instancetype: 'coursecommunication',
+            instanceid: $courseid,
+            provider: 'communication_matrix',
+        );
+```
+
+### Important features of the API
+
+The below are the important methods/features of the API. These methods/features are the ones which are used by the hook listeners to interact with the communication API.
+
+#### Create and configure a room
+
+$communication->create_and_configure_room() method is used to add an ad-hoc to create a room in the communication provider and configure it with the required settings.
+
+```php
+public function create_and_configure_room(
+        string $communicationroomname,
+        ?\stored_file $avatar = null,
+        ?\stdClass $instance = null,
+    )
+```
+
+This method takes the following parameters:
+
+- The name of the room as a string to be created.
+- The avatar of the room to be created, this is a stored file object.
+- The instance object of the communication API.
+
+For example, we want to create a room for a course with the course name as the room name and the course image as the avatar, we can use the below code.
+There can be other cases where information from course instance will be used by a plugin, for example, the dynamic field from matrix plugin has a form
+field named topic which sets the topic of the room, the instance object can be used to get the topic from the course instance and set it in the form field.
+Please refer to [Communication plugin](../../plugintypes/communication/index.md) documentation for more details.
+
+```php
+// first initialize the communication instance, provided the context is already initialized and the selected provider is available though a form or other means.
```

Comment:
```suggestion
// First initialize the communication instance, provided the context is already initialized and the selected provider is available
// though a form or other means.
```

Reply:
The given examples should follow the coding style rules (because they will be taken as a good practices to follow) ;-)